File: BBB.Framework.Dto/Core/ParticipantDto.cs
namespace BBB.Framework.Dto.Core
{
    public sealed class ParticipantDto
    {
        public Guid ParticipantKey { get; set; }

        public Guid PlayerKey { get; set; }

        public PlayerDto? Player { get; set; }

        public Guid ScheduleKey { get; set; }

        public List<EncounterPositionParticipantDto> EncounterPositions { get; set; } = new();
    }
}

File: BBBRaiderWeb/src/models/ParticipantDto.ts
import { logTypeMissmatch, parseString } from '../helpers/model'
import { PlayerDto } from './PlayerDto'

export class ParticipantDto {
  participantKey: string = ''
  playerKey: string = ''
  player?: PlayerDto
  scheduleKey: string = ''

  static fromJSON(obj: any): ParticipantDto {
    if (!obj) {
      logTypeMissmatch(`ParticipantDto`, obj)
      return new ParticipantDto()
    }

    return {
      participantKey: parseString(obj.participantKey),
      playerKey: parseString(obj.playerKey),
      player: PlayerDto.fromJSON(obj.player),
      scheduleKey: parseString(obj.scheduleKey),
    }
  }
}
